strawberry/exceptions/invalid_argument_type.py

```diff
@@ -0,0 +1,50 @@
+from typing import TYPE_CHECKING, Optional
+
+from typing_extensions import Literal
+
+from .exception import StrawberryException
+from .exception_source import ExceptionSource
+from .utils.source_finder import SourceFinder
+
+
+if TYPE_CHECKING:
+    from strawberry.types.fields.resolver import StrawberryResolver
+
+
+class InvalidArgumentTypeError(StrawberryException):
+    def __init__(
+        self,
+        resolver: "StrawberryResolver",
+        argument_name: str,
+        argument_type: Literal["union", "interface"],
```

Comment:
perhaps this should be an enum? Or maybe you should just pass the `StrawberryArgument` and hte exception can grab `argument.type`?

Reply:
that's a good suggestion, I'll see if we can use StrawberryArgument, if not I'll go with enum 😊